qiskit/pulse/timeslots.py

```diff
@@ -217,36 +217,47 @@ def ch_stop_time(self, *channels: List[Channel]) -> int:
             return max(interval.end for interval in intervals)
         return 0
 
-    def ch_duration(self, *channels: List[Channel]) -> int:
+    def ch_duration(self, *channels: Channel) -> int:
         """Return maximum duration of timeslots over all channels.
 
         Args:
             *channels: Channels over which to obtain the duration.
         """
         return self.ch_stop_time(*channels)
 
-    def is_mergeable_with(self, timeslots: 'TimeslotCollection') -> bool:
-        """Return if self is mergeable with `timeslots`.
+    def is_mergeable_with(self, other: 'TimeslotCollection') -> bool:
+        """Return if self is mergeable with `other` collection.
 
         Args:
-            timeslots: TimeslotCollection to be checked
+            other: TimeslotCollection to be checked
         """
-        for slot in timeslots.timeslots:
+        for slot in other.timeslots:
             if slot.channel in self.channels:
```

Comment:
So, the inefficiency wasn't in checking for overlap, it was just in constructing new Timeslots?

Reply:
That's right. The bottleneck was in the construction of new Timesltos.

Reply:
Yes, not in checking for overlap. The inefficiency was in the full reconstruction of _table (and _timeslots) in constructing a new TimeslotCollection.